oaknational/Oak-Web-Application
# Allow openapi backend with `ENABLE_OPEN_API` + env variables

## Description
This PR adds an openapi backend to the curriculum visualiser. You can enable this by

 - Set `ENABLE_OPEN_API=true` in `src/utils/curriculum/constants.ts`
 - Set `OPENAPI_KEY` to your personal open API key
 - Set `OPENAPI_URL` to the instance of the running openapi server
   - Currently supported server is `OPENAPI_URL="https://oak-openapi-1ncwpjnwi.vercel-preview.thenational.academy"` 

## Issue(s)

Fixes `CUR-1120`

## How to test
With `ENABLE_OPEN_API` disabled

1. Go to https://deploy-preview-3077--oak-web-application.netlify.thenational.academy/teachers/curriculum
2. Make sure it's requesting from the curric API and not the openapi
